iml-api/src/graphql.rs

```diff
@@ -141,6 +142,86 @@ impl Deref for SortDir {
     }
 }
 
+#[derive(GraphQLEnum, sqlx::Type)]
+#[repr(i16)]
+enum MessageClass {
+    Normal = 0,
+    Lustre = 1,
+    LustreError = 2,
+    Copytool = 3,
+    CopytoolError = 4,
+}
+
+impl From<i16> for MessageClass {
+    fn from(value: i16) -> Self {
+        match value {
+            0 => MessageClass::Normal,
+            1 => MessageClass::Lustre,
+            2 => MessageClass::LustreError,
+            3 => MessageClass::Copytool,
+            4 => MessageClass::CopytoolError,
+            _ => panic!("Invalid variant"),
+        }
+    }
+}
+
+#[derive(GraphQLEnum, sqlx::Type)]
+#[repr(i16)]
+enum LogSeverity {
+    Emergency = 0,
+    Alert = 1,
+    Critical = 2,
+    Error = 3,
+    Warning = 4,
+    Notice = 5,
+    Informational = 6,
+    Debug = 7,
+}
+
+impl From<i16> for LogSeverity {
+    fn from(value: i16) -> Self {
```

Comment:
I think using `TryFrom` would be better than panicking if we can't find a conversion.